qemu.go

```diff
@@ -976,22 +1002,30 @@ func (q *qemu) hotplugBlockDevice(drive Drive, op operation) error {
 			return err
 		}
 
-		driver := "virtio-blk-pci"
+		if q.config.BlockDeviceDriver == VirtioBlock {
+			driver := "virtio-blk-pci"
+			addr, bus, err := q.addDeviceToBridge(drive.ID)
 
-		addr, bus, err := q.addDeviceToBridge(drive.ID)
-		if err != nil {
-			return err
-		}
+			if err = q.qmpMonitorCh.qmp.ExecutePCIDeviceAdd(q.qmpMonitorCh.ctx, drive.ID, devID, driver, addr, bus); err != nil {
+				return err
+			}
+		} else {
+			driver := "scsi-hd"
 
-		if err = q.qmpMonitorCh.qmp.ExecutePCIDeviceAdd(q.qmpMonitorCh.ctx, drive.ID, devID, driver, addr, bus); err != nil {
-			return err
-		}
+			// Bus exposed by the SCSI Controller
+			bus := scsiControllerID + ".0"
 
-	} else {
-		if err := q.removeDeviceFromBridge(drive.ID); err != nil {
-			return err
-		}
+			// Get SCSI-id and LUN based on the order of attaching drives.
+			scsiID, lun, err := getSCSIIdLun(drive.Index)
+			if err != nil {
+				return err
+			}
 
+			if err = q.qmpMonitorCh.qmp.ExecuteSCSIDeviceAdd(q.qmpMonitorCh.ctx, drive.ID, devID, driver, bus, scsiID, lun); err != nil {
```

Comment:
The code looks good but I have noticed that you're not handling the case where we want to remove a SCSI device. Am I missing something about the way SCSI works, or have you forgotten to implement this ?

Reply:
The way to remove SCSI device is the same as virtio-block device.

Reply:
Makes sense, thanks !